terraform/full_environment/events_ingestion.tf

```diff
@@ -0,0 +1,188 @@
+# service account for reading data
+resource "google_service_account" "analytics_write" {
+  account_id   = "ga4-write-bq"
+  display_name = "ga4-write-bq"
+  project      = var.gcp_project_id
+}
+
+# custom role for analytics transfer
+resource "google_project_iam_custom_role" "analytics_write_role" {
+  role_id     = "analytics_write_role"
+  title       = "ga4-write-bq-permissions"
+  description = "Write data to vertex schemas in bq"
+
+  permissions = [
+    "bigquery.tables.update",
+    "bigquery.tables.updateData",
+    "bigquery.jobs.create"
+  ]
+}
+
+# binding role to service account
+resource "google_project_iam_binding" "analytics_write" {
+  project = var.gcp_project_id
+  role    = google_project_iam_custom_role.analytics_write_role.id
+
+  members = [
+    google_service_account.analytics_write.member
+  ]
+}
+
+###
+
+# service account for reading ga4 data
+resource "google_service_account" "analytics_read" {
```

Comment:
**Note to self (@csutter):** This will need ID federation in the source project, I'll try and get that set up today

Reply:
Note to self again: this will be sorted out for us in the source project